src/gui/transferlistfilterswidget.cpp

```diff
@@ -445,41 +447,38 @@ void TrackerFiltersList::refreshTrackers(const BitTorrent::Torrent *torrent)
 void TrackerFiltersList::changeTrackerless(const BitTorrent::Torrent *torrent, const bool trackerless)
 {
     if (trackerless)
-        addItem(NULL_HOST, torrent->id());
+        addItems(NULL_HOST, {torrent->id()});
     else
         removeItem(NULL_HOST, torrent->id());
 }
 
-void TrackerFiltersList::addItem(const QString &tracker, const BitTorrent::TorrentID &id)
+void TrackerFiltersList::addItems(const QString &trackerURL, const QVector<BitTorrent::TorrentID> &torrents)
 {
-    const QString host = getHost(tracker);
+    const QString host = getHost(trackerURL);
     auto trackersIt = m_trackers.find(host);
     const bool exists = (trackersIt != m_trackers.end());
     QListWidgetItem *trackerItem = nullptr;
 
     if (exists)
     {
-        if (trackersIt->torrents.contains(id))
-            return;
-
         trackerItem = trackersIt->item;
     }
     else
     {
         trackerItem = new QListWidgetItem();
         trackerItem->setData(Qt::DecorationRole, UIThemeManager::instance()->getIcon(u"network-server"_qs));
 
-        TrackerData trackerData {{}, trackerItem};
+        const TrackerData trackerData {{}, trackerItem};
         trackersIt = m_trackers.insert(host, trackerData);
 
-        const QString scheme = getScheme(tracker);
+        const QString scheme = getScheme(trackerURL);
         downloadFavicon(u"%1://%2/favicon.ico"_qs.arg((scheme.startsWith(u"http") ? scheme : u"http"_qs), host));
     }
 
     Q_ASSERT(trackerItem);
 
     QSet<BitTorrent::TorrentID> &torrentIDs = trackersIt->torrents;
-    torrentIDs.insert(id);
+    torrentIDs.unite({torrents.cbegin(), torrents.cend()});
```

Comment:
Might be faster to use a loop directly.
Currently it needs to construct a QSet first and then throw it away after use.

Reply:
> Might be faster to use a loop directly.
> Currently it needs to construct a QSet first and then throw it away after use.

Changed.